src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    this.forEach((e, i) => {
```

Comment:
```suggestion
    this.forEach((e, i) => {
```
Let's try not to use another array methods while implementing this one. Use simple `for` loop here